quantum/process_keycode/process_combo.c

```diff
@@ -549,6 +556,13 @@ bool process_combo(uint16_t keycode, keyrecord_t *record) {
 #ifdef COMBO_ONLY_FROM_LAYER
     /* Only check keycodes from one layer. */
     keycode = keymap_key_to_keycode(COMBO_ONLY_FROM_LAYER, record->event.key);
+#else
+
+    uint8_t highest_layer = get_highest_layer(layer_state);
+    layer                 = combo_ref_from_layer(highest_layer);
+    if (layer != highest_layer) {
+        keycode = keymap_key_to_keycode(layer, record->event.key);
+    }
```

Comment:
Looks like the `layer` variable is used only by this part of code, so you may as well declare it here (and maybe rename to `ref_layer` to be more descriptive):
```suggestion
    uint8_t highest_layer = get_highest_layer(layer_state);
    uint8_t ref_layer     = combo_ref_from_layer(highest_layer);
    if (ref_layer != highest_layer) {
        keycode = keymap_key_to_keycode(ref_layer, record->event.key);
    }
```